build.gradle

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.10'
+    }
+}
+
+allprojects {
+    apply from: "$rootDir/dependencies.gradle"
+
```

Comment:
The build doesn't currently specify Java source compatibility and target compatibility, which is necessary.
Specifying the common settings for all Java projects could be done here, for example:
```
    pluginManager.withPlugin('java') {
        sourceCompatibility = targetCompatibility = 8
    }
```

Reply:
I would probably not vote for tying to any version and use as default. Which essentially means 

- sourceCompatibility will be the version of JVM. 
- targetCompatibility will be same as sourceCompatibility 

Therefore If anybody wants certain build and source version adjust the JVM version.

Reply:
That's not a very common solution to let it float. Usually a project sets the sourceCompatibilty and targetCompatibility to a specific version. In a OSS library project like BookKeeper, I don't think it's a good idea to not specify it.
Even if you set sourceCompatibility and targetCompatibility to Java 8, there will be a benefit in using Java 11. 
Since Gradle 6.6 there is also support for passing the "release" flag to the compiler, docs in https://docs.gradle.org/6.6/release-notes.html#javacompile-release .
Using the release flag is necessary when building on newer JVMs and targeting Java 8 for preventing issues such as https://github.com/apache/pulsar/issues/8445 .
If there's a need to easily change the version for custom forks etc, the logic could be made configurable/conditional with a `gradle.properties` property.

Reply:
If that's the conventional wisdom then I will accept this.

Reply:
the comments I made in the review weren't meant to block the work, but provide some feedback on what common Gradle build best practices are. Feel free to ignore my comments. :)

Reply:
@lhotari Your comments were invaluable. And I really appreciate that. Keep them coming. This will be the first commit for gradle and it's important that we start with the right set or rules and practices.

Reply:
Thanks, @lhotari , and as @pkumar-singh said, there can and will be follow ups to make the Gradle solution better.
This PR will be the "bootstrapping" effort for us to add the improvements to keep us moving forward.

Reply:
yes it's fine to get this PR merged and gradually improve the Gradle build. The current build does have issues with up-to-date checks since all tasks don't properly declare inputs and outputs. This is the reason why "gradle clean" will be occasionally needed. In that sense this build has bugs. Once the build is merged, it's easier for me to contribute fixes for this.